sparkle/src/components/Markdown.tsx

```diff
@@ -1,13 +1,41 @@
-import React from "react";
+import React, { useCallback, useState } from "react";
 import ReactMarkdown from "react-markdown";
 
+import { ClipboardCheckIcon, ClipboardIcon, IconButton } from "@sparkle/index";
+
 export function Markdown({
   content,
   className = "",
 }: {
   content: string;
   className?: string;
 }) {
+  function useCopyToClipboard(
+    resetInterval = 2000
+  ): [isCopied: boolean, copy: (d: ClipboardItem) => Promise<boolean>] {
+    const [isCopied, setCopied] = useState(false);
+
+    const copy = useCallback(
```

Comment:
same with useCallback, it's used to memoize a function across renders, but here it won't be run at each render so it might lead to issues

Reply:
Same comment as up, this is the same code we use on Front for the conversation component

Reply:
What are the issues it can lead to?